pkg/openstack/openstack_test.go

```diff
@@ -878,6 +878,233 @@ func TestNodeAddressesWithAddressSortOrderOptions(t *testing.T) {
 	}
 }
 
+func TestNodeAddressesWithSubports(t *testing.T) {
```

Comment:
This isn't really testing the interesting code. In fact, nothing here necessarily relates to subports. We need test coverage for getAttachedInterfacesByID(), e.g.:

* Trunking is disabled
* Trunking is enabled but there are no trunks
* A single port has a single subport
* A single port has multiple subports
* Multiple ports have multiple subports

Reply:
hmm...I will get back for this.

Reply:
sure, will get resources to add those unit tests